packages/flutter/test/material/reorderable_list_test.dart

```diff
@@ -1259,6 +1259,68 @@ void main() {
     });
     // TODO(djshuckerow): figure out how to write a test for scrolling the list.
   });
+
+  testWidgets('ReorderableListView, can deal with the dragged item getting unmounted and rebuilt during drag', (WidgetTester tester) async {
+    // See https://github.com/flutter/flutter/issues/74840 for more details.
+    final List<int> items = List<int>.generate(100, (int index) => index);
+
+    void handleReorder(int fromIndex, int toIndex) {
+      if (toIndex > fromIndex) {
+        toIndex -= 1;
+      }
+      items.insert(toIndex, items.removeAt(fromIndex));
+    }
+
+    // The list is 800x600, 8 items, each item is 800x100 with
+    // an "item $index" text widget at the item's origin.  Drags are initiated by
+    // a simple press on the text widget.
+    await tester.pumpWidget(MaterialApp(
+      home: ReorderableListView.builder(
+        itemBuilder: (BuildContext context, int index) {
+          return Container(
+            key: ValueKey<int>(items[index]),
+            height: 100,
+            child: ReorderableDragStartListener(
+              child: Row(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: <Widget>[
+                  Text('item ${items[index]}'),
+                ],
+              ),
+              index: index,
+            ),
+          );
+        },
+        itemCount: items.length,
+        onReorder: handleReorder,
+      ),
+    ));
+
+    // Drag item 0 downwards and force an auto scroll off the end of the list
+    // far enough that item zeros original entry in the list is unmounted.
+    final TestGesture drag = await tester.startGesture(tester.getCenter(find.text('item 0')));
+    await tester.pump(kPressTimeout);
+    // Off the bottom of the screen, which should autoscroll until we hit the
+    // end of the list
+    await drag.moveBy(const Offset(0, 700));
+    await tester.pump(const Duration(seconds: 30));
+    await tester.pumpAndSettle();
+    // Drag back to off the top of the list, which should autoscroll until
```

Comment:
Should confirm that the last 6 items are visible here, i.e. we _really_ autoscrolled to the end.

Reply:
Good idea. Thx.